data.tf

```diff
@@ -0,0 +1,14 @@
+data "azurerm_client_config" "current" {}
+data "azurerm_dns_zone" "zone" {
+  provider            = azurerm.dnszone
+  for_each            = toset(var.dns_zones)
+  name                = each.value
+  resource_group_name = var.dns_zone_rg
+}
+data "azurerm_subscription" "subid" {
+  provider = "azurerm.dnszone"
+}
+
+data "azuread_groups" "groups" {
```

Comment:
you don't need to use groups when only looking up one group, just use `azuread_group`